src/Relation/Nullary/Construct/Add/Point.agda

```diff
@@ -8,8 +8,84 @@
 
 module Relation.Nullary.Construct.Add.Point where
 
+open import Algebra.Bundles
+open import Algebra.Core using (Op₂)
+import Algebra.Definitions as Definitions
+import Algebra.Structures as Structures
+open import Level using (Level)
 open import Data.Maybe.Base public
   using () renaming (Maybe to Pointed; nothing to ∙; just to [_])
 
 open import Data.Maybe.Properties public
   using (≡-dec) renaming (just-injective to []-injective)
+open import Data.Product using (_,_)
+open import Relation.Binary.PropositionalEquality
+
+private
+  variable
+    a : Level
+
+module _ {A : Set a} {op : Op₂ A} where
+  open Definitions {A = Pointed A} _≡_
```

Comment:
Using propositional equality everywhere is a bit restrictive. I think we should take the underlying equality of the structure without the point, and then use pointed equality defined in `Relation.Binary.Construct.Add.Point` as the equality for the new structure?